cpp/src/strings/convert/convert_datetime.cu

```diff
@@ -416,18 +439,23 @@ struct check_datetime_format {
    * @param bytes Number of bytes to check.
    * @param min_value Inclusive minimum value
    * @param max_value Inclusive maximum value
-   * @return true if parsed value is between `min_value` and `max_value`.
+   * @return number of bytes not successfully processed
    */
-  __device__ bool check_value(const char* str, size_type bytes, int min_value, int max_value)
+  __device__ size_type check_value(const char* str,
```

Comment:
Feels smelly to return only bytes for a function that's supposed to do a checking. Consider returning pair<bool, size_type>.